industrial_robot_client/src/joint_trajectory_action.cpp

```diff
@@ -77,6 +80,29 @@ void JointTrajectoryAction::robotStatusCB(const industrial_msgs::RobotStatusCons
   has_moved_once_ = has_moved_once_ ? true : (last_robot_status_->in_motion.val == industrial_msgs::TriState::TRUE);
 }
 
+
+bool JointTrajectoryAction::disableRobotCB(std_srvs::SetBool::Request &req,
+                                         std_srvs::SetBool::Response &res)
+{
+  res.success = req.data != disabled_;
+  disabled_ = req.data;
+
+  if (disabled_ && has_active_goal_)
+    cancelCB(active_goal_);
+  
+  if (!res.success)
+    res.message="Robot was already ";
+  else
+    res.message="Robot is now ";
+
+  if (disabled_)
+    res.message += "disabled";
+  else res.message += "enabled";
+  
+  return true;
+  
+}
```

Comment:
This is again more of a question than a request (so feel free to disagree / tell me to do this myself), but would it be nice to split this into a `disable_robot` and an `enable_robot` service (and use `std_srvs/Trigger` for the type)?

It would simplify this code and be very explicit from a component interface perspective.

Finally: should a getter service be added as well? That way we can retrieve the current state.

Reply:
I guess I did this split a long time ago.  There is no getter, but with a ROS_WARN when you try to send after disabling, it should be obvious.  And there's no reason why you couldn't enable or disable multiple times if needed.

Reply:
Yes.  It has been split and there is a getter.